### PR TITLE
Changelog 3.7.00

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
First pass at the changelog, please push any desired updates to curate

File: CMakeLists.txt [hunk 1]
```diff
@@ -128,8 +128,8 @@ ENDIF()
 
 
 set(Kokkos_VERSION_MAJOR 3)
-set(Kokkos_VERSION_MINOR 6)
-set(Kokkos_VERSION_PATCH 99)
+set(Kokkos_VERSION_MINOR 7)
+set(Kokkos_VERSION_PATCH 00)
```
Comment (dalg24) [inline CMakeLists.txt:132]: Sorry I merged #5358 before I saw this

**Review by crtrott — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,176 @@
...
+- Add `is_finalized()` [\#5247](https://github.com/kokkos/kokkos/pull/5247)
+- Promote mathematical functions to Kokkos namespace [\#4791](https://github.com/kokkos/kokkos/pull/4791)
+- Promote `min`, `max`, `clamp`, `minmax` functions to Kokkos namespace [\#5170](https://github.com/kokkos/kokkos/pull/5170)
+- Add `round`, `logb`, `nextafter`, `copysign`, and `signbit` math functions [\#4768](https://github.com/kokkos/kokkos/pull/4768)
+- Add `HIPManagedSpace`, similar to `CudaUVMSpace` [\#5112](https://github.com/kokkos/kokkos/pull/5112)
+- Accept view construction allocation properties in `create_mirror[_view,_view_and_copy]` and `resize/realloc` [\#5125](https://github.com/kokkos/kokkos/pull/5125) [\#5095](https://github.com/kokkos/kokkos/pull/5095) [\#5035](https://github.com/kokkos/kokkos/pull/5035) [\#4805](https://github.com/kokkos/kokkos/pull/4805) [\#4844](https://github.com/kokkos/kokkos/pull/4844)
+- Allow `MemorySpace::allocate()` to be called with execution space [\#4826](https://github.com/kokkos/kokkos/pull/4826)
+- Experimental: Compile time view subscriber and dynamic view hooks [\#4197](https://github.com/kokkos/kokkos/pull/4197)
```
Comment (JBludau) [inline CHANGELOG.md:16]: Do we want to specifically name the new `is_finalized()` function?

**Review by JBludau — APPROVED** [review]
looks good

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,176 @@
...
+- Add arch flags for Intel GPU Ponte Vecchio [\#4932](https://github.com/kokkos/kokkos/pull/4932)
+- SYCL: require GPU if GPU architecture was set at configuration time (i.e. do not allow fallback to CPU device) [\#5264](https://github.com/kokkos/kokkos/pull/5264) [\#5222](https://github.com/kokkos/kokkos/pull/5222)
+- SYCL: Add `SYCL::sycl_queue()` for interoperability [\#5241](https://github.com/kokkos/kokkos/pull/5241)
+- SYCL: Loosen restriction for using built-in `sycl::group_broadcast` [\#4552](https://github.com/kokkos/kokkos/pull/4552)
+- SYCL: preserve address space [\#4396](https://github.com/kokkos/kokkos/pull/4396)
+- OpenMPTarget: Adding a workaound for team scan [\#5219](https://github.com/kokkos/kokkos/pull/5219)
+- OpenMPTarget: Adding logic to skip the kernel launch if `league_size=0` [\#5067](https://github.com/kokkos/kokkos/pull/5067)
+- OpenMPTarget: Make sure `Kokkos::abort()` causes abnormal program termination when called on the host-side [\#4808](https://github.com/kokkos/kokkos/pull/4808)
+- HIP: Make HIPHostPinnedSpace goarse grain [\#5152](https://github.com/kokkos/kokkos/pull/5152)
```
Comment (masterleinad) [inline CHANGELOG.md:29]:
```suggestion
- HIP: Make HIPHostPinnedSpace coarse-grained [\#5152](https://github.com/kokkos/kokkos/pull/5152)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,176 @@
...
+- Update numerics support for `__float128` [\#5081](https://github.com/kokkos/kokkos/pull/5081)
+- Add `log10` overload for `Kokkos::complex` [\#5009](https://github.com/kokkos/kokkos/pull/5009)
+- Add `[[nodiscard]]` to `ScopeGuard` [\#5224](https://github.com/kokkos/kokkos/pull/5224)
+- Add structured binding support for `Kokkos::Array` [\#4962](https://github.com/kokkos/kokkos/pull/4962)
+- Enable accessing `Kokkos::Array` elements in constant expressions [\#4916](https://github.com/kokkos/kokkos/pull/4916)
+- Mark `as_view_of_rank_n` as KOKKOS_FUNCTION [\#5248](https://github.com/kokkos/kokkos/pull/5248)
+- Cleanup/rework fence overloads [\#5148](https://github.com/kokkos/kokkos/pull/5148)
+- Assert that `Layout` construction from extents is valid in functions taking integer extents [\#5209](https://github.com/kokkos/kokkos/pull/5209)
+- Add `fill_random` overload that takes an execution space as first argument [\#5181](https://github.com/kokkos/kokkos/pull/5181)
+- Remove useless `memcpy` [\#5161](https://github.com/kokkos/kokkos/pull/5161)
```
Comment (masterleinad) [inline CHANGELOG.md:48]: This only modifies code in a test.
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,176 @@
...
+- Print help only once [\#5128](https://github.com/kokkos/kokkos/pull/5128)
+- Update precedence rule in initialization [\#5130](https://github.com/kokkos/kokkos/pull/5130)
+- Warn instead of just ignoring user settings when kokkos-tools is disabled [\#5088](https://github.com/kokkos/kokkos/pull/5088)
+- Drop numa args in threads backend initialization [\#5127](https://github.com/kokkos/kokkos/pull/5127)
+- Warn users when a flag prefixed with -[-]kokkos is not recognized and do not remove it [\#5256](https://github.com/kokkos/kokkos/pull/5256)
+- Give back to Core what belongs to Core (aka moving tune_internals option from Tools back to Core) [\#5202](https://github.com/kokkos/kokkos/pull/5202)
+
+#### Build system updates:
+- Support finding libquadmath with native compiler support [\#5286](https://github.com/kokkos/kokkos/pull/5286)
+- Enable using C++23 [\#5283](https://github.com/kokkos/kokkos/pull/5283)
```
Comment (masterleinad) [inline CHANGELOG.md:73]: I don't think these are part of the 3.7 release.
```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,176 @@
 # Change Log
 
+## [3.7.00](https://github.com/kokkos/kokkos/tree/3.7.00) (2022-08-18)
+[Full Changelog](https://github.com/kokkos/kokkos/compare/3.6.01...3.7.09)
```
Comment (cz4rs) [inline CHANGELOG.md:4]: Typo in the version number?
```suggestion
[Full Changelog](https://github.com/kokkos/kokkos/compare/3.6.01...3.7.00)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,176 @@
 # Change Log
 
+## [3.7.00](https://github.com/kokkos/kokkos/tree/3.7.00) (2022-08-18)
```
Comment (cz4rs) [inline CHANGELOG.md:3]: Does the date matter?
```suggestion
## [3.7.00](https://github.com/kokkos/kokkos/tree/3.7.00) (2022-08-22)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,176 @@
...
+- Disallow multiple host architectures in CMake [\#4996](https://github.com/kokkos/kokkos/pull/4996)
+- Do not include compiler warning flags in the compile option of the cmake target [\#4989](https://github.com/kokkos/kokkos/pull/4989)
+- AOT flags for OpenMPTarget targeting Intel GPUs [\#4915](https://github.com/kokkos/kokkos/pull/4915)
+- Repurpose `Kokkos_ARCH_INTEL_GEN` for SYCL to mean JIT to be conforming with OMPT [\#4894](https://github.com/kokkos/kokkos/pull/4894)
+- Replace amdgpu-target with offload-arch [\#4874](https://github.com/kokkos/kokkos/pull/4874)
+- Do not enable `kokkos_launch_compiler` when `CMAKE_CXX_COMPILER_LAUNCHER` is set [\#4870](https://github.com/kokkos/kokkos/pull/4870)
+- Move CMake version check up [\#4797](https://github.com/kokkos/kokkos/pull/4797)
+
+### Incompatibilities:
+- Turn setting Kokkos_CXX_STANDARD into an error [\#5293](https://github.com/kokkos/kokkos/pull/5293)
```
Comment (masterleinad) [inline CHANGELOG.md:90]: also only on `develop`.
```suggestion
```

File: CHANGELOG.md (hunk 2 above)
Comment (nmm0) [inline CHANGELOG.md:16]:
```suggestion
- Experimental: Compile time view subscriber [\#4197](https://github.com/kokkos/kokkos/pull/4197)
```

Dynamic view hooks was pulled out of the PR before merging, sorry I didn't update the name

**Review by nmm0 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (crtrott) [reply CHANGELOG.md:16]: yes

**Review by crtrott — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (crtrott) [reply CHANGELOG.md:73]: yeah you are right

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,172 @@
 # Change Log
 
+## [3.7.00](https://github.com/kokkos/kokkos/tree/3.7.00) (2022-08-22)
+[Full Changelog](https://github.com/kokkos/kokkos/compare/3.6.01...3.7.00)
+
+### Features:
+- Use non-volatile `join()` routines in `parallel_reduce/scan` [\#4931](https://github.com/kokkos/kokkos/pull/4931) [\#4954](https://github.com/kokkos/kokkos/pull/4954) [\#4951](https://github.com/kokkos/kokkos/pull/4951)
```
Comment (PhilMiller) [inline CHANGELOG.md:7]:
```suggestion
- Use non-volatile `join()` and `operator+=` routines in `parallel_reduce/scan` [\#4931](https://github.com/kokkos/kokkos/pull/4931) [\#4954](https://github.com/kokkos/kokkos/pull/4954) [\#4951](https://github.com/kokkos/kokkos/pull/4951)
```

Not critical, but it is salient I think

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,172 @@
...
+- Add `SIMD` sub package (requires C++17) [\#5016](https://github.com/kokkos/kokkos/pull/5016)
+- Add `is_finalized()` [\#5247](https://github.com/kokkos/kokkos/pull/5247)
+- Promote mathematical functions to Kokkos namespace [\#4791](https://github.com/kokkos/kokkos/pull/4791)
+- Promote `min`, `max`, `clamp`, `minmax` functions to Kokkos namespace [\#5170](https://github.com/kokkos/kokkos/pull/5170)
```
Comment (PhilMiller) [inline CHANGELOG.md:11]:
```suggestion
- Promote mathematical functions from `namespace Kokkos::Experimental` to `namespace Kokkos` [\#4791](https://github.com/kokkos/kokkos/pull/4791)
- Promote `min`, `max`, `clamp`, `minmax` functions from `namespace Kokkos::Experimental` to `namespace Kokkos` [\#5170](https://github.com/kokkos/kokkos/pull/5170)
```

**Review by nmm0 — APPROVED** [review]
Looks good to me, once other comments are adressed

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,172 @@
...
+- SYCL: Add `SYCL::sycl_queue()` for interoperability [\#5241](https://github.com/kokkos/kokkos/pull/5241)
+- SYCL: Loosen restriction for using built-in `sycl::group_broadcast` [\#4552](https://github.com/kokkos/kokkos/pull/4552)
+- SYCL: preserve address space [\#4396](https://github.com/kokkos/kokkos/pull/4396)
+- OpenMPTarget: Adding a workaound for team scan [\#5219](https://github.com/kokkos/kokkos/pull/5219)
+- OpenMPTarget: Adding logic to skip the kernel launch if `league_size=0` [\#5067](https://github.com/kokkos/kokkos/pull/5067)
+- OpenMPTarget: Make sure `Kokkos::abort()` causes abnormal program termination when called on the host-side [\#4808](https://github.com/kokkos/kokkos/pull/4808)
+- HIP: Make HIPHostPinnedSpace coarse-grained [\#5152](https://github.com/kokkos/kokkos/pull/5152)
+- Refactor OpenMP `parallel_for` implementation to use more native OpenMP constructs [\#4664](https://github.com/kokkos/kokkos/pull/4664)
+- Add option to optimize for local CPU architecture `Kokkos_ARCH_NATIVE` [\#4930](https://github.com/kokkos/kokkos/pull/4930)
```
Comment (PhilMiller) [inline CHANGELOG.md:31]: I'd suggest moving these two up to the top of this section, as probably the most salient for many users

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,172 @@
...
+- Deprecate command line arguments (other than --help) that are not prefixed with --kokkos* [\#5120](https://github.com/kokkos/kokkos/pull/5120)
+- Deprecate --[kokkos-]numa cmdline arg and KOKKOS_NUMA env var [\#5117](https://github.com/kokkos/kokkos/pull/5117)
+- Deprecate --[kokkos-]threads command line argument in favor of --[kokkos-]num-threads [\#5111](https://github.com/kokkos/kokkos/pull/5111)
+- Deprecate Kokkos::common_view_alloc_prop [\#5059](https://github.com/kokkos/kokkos/pull/5059)
+- Deprecate Kokkos::is_reducer_type [\#4957](https://github.com/kokkos/kokkos/pull/4957)
+- Deprecate OffsetView constructors taking index_list_type [\#4810](https://github.com/kokkos/kokkos/pull/4810)
+- Warn about parallel_reduce cases that call join() with volatile-qualified arguments [\#5215](https://github.com/kokkos/kokkos/pull/5215)
+
+### Bug Fixes:
+- CUDA Reductions: Fix data races reported [\#4855](https://github.com/kokkos/kokkos/pull/4855)
```
Comment (PhilMiller) [inline CHANGELOG.md:111]:
```suggestion
- CUDA Reductions: Fix data races reported by Nvidia `compute-sanitizer` [\#4855](https://github.com/kokkos/kokkos/pull/4855)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,172 @@
...
+- Work around intel compiler bug [\#5301](https://github.com/kokkos/kokkos/pull/5301)
```
Comment (PhilMiller) [inline CHANGELOG.md:112]:
```suggestion
- Work around Intel compiler bug [\#5301](https://github.com/kokkos/kokkos/pull/5301)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,172 @@
...
+- Avoid allocating memory for UniqueToken [\#5300](https://github.com/kokkos/kokkos/pull/5300)
+- DynamicView: Properly resize mirror instances after construction [\#5276](https://github.com/kokkos/kokkos/pull/5276)
+- Remove Kokkos::Rank limit of 6 ranks [\#5271](https://github.com/kokkos/kokkos/pull/5271)
+- Do not forget to set last element to nullptr when removing a flag in Kokkos::initialize [\#5272](https://github.com/kokkos/kokkos/pull/5272)
+- Fix CUDA+MSVC build issue [\#5261](https://github.com/kokkos/kokkos/pull/5261)
+- Fix DynamicView::resize_serial [\#5220](https://github.com/kokkos/kokkos/pull/5220)
+- Fix cmake default compiler flags for unknown compiler [\#5217](https://github.com/kokkos/kokkos/pull/5217)
+- Fix move backward [\#5191](https://github.com/kokkos/kokkos/pull/5191)
```
Comment (PhilMiller) [inline CHANGELOG.md:120]:
```suggestion
- Fix `move_backward` [\#5191](https://github.com/kokkos/kokkos/pull/5191)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,172 @@
...
+- Guard against unrecognized pragma with intel compilers [\#5019](https://github.com/kokkos/kokkos/pull/5019)
+- Fix racing condition in HIPParallelLaunch [\#5008](https://github.com/kokkos/kokkos/pull/5008)
+- KokkosP: Fix device_id in profiling [\#4997](https://github.com/kokkos/kokkos/pull/4997)
+- Fix for Kokkos::vector::insert into empty vector with begin and end iterators [\#4988](https://github.com/kokkos/kokkos/pull/4988)
+- Fix Core header files installation [\#4984](https://github.com/kokkos/kokkos/pull/4984)
+- Fix bounds errors with Kokkos::sort [\#4980](https://github.com/kokkos/kokkos/pull/4980)
+- Fixup let RangePolicy::set_chunk_size return a reference to self [\#4918](https://github.com/kokkos/kokkos/pull/4918)
+- Fix allocating large Views [\#4907](https://github.com/kokkos/kokkos/pull/4907)
+- Fix combined reductions with Kokkos::View [\#4896](https://github.com/kokkos/kokkos/pull/4896)
+- Fixed _CUDA_ARCH__ to __CUDA_ARCH__ for CUDA LDG [\#4893](https://github.com/kokkos/kokkos/pull/4893)
```
Comment (PhilMiller) [inline CHANGELOG.md:142]:
```suggestion
- Fixed `_CUDA_ARCH__` to `__CUDA_ARCH__` for CUDA LDG [\#4893](https://github.com/kokkos/kokkos/pull/4893)
```

**Comment by PhilMiller** [comment]
It would be nice if every code-like symbol were formatted as `code` but I only think the one where markdown produces other formatting is really essential.

File: CHANGELOG.md (hunk 11 above)
Comment (ajpowelsnl) [reply CHANGELOG.md:31]: Why abbreviate "architecture" (with `Arch`) in a section title?

**Review by PhilMiller — APPROVED** [review]

**Review by ldh4 — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**